app/v2/connectors/V2BaseConnector.scala

```diff
@@ -105,17 +114,21 @@ trait V2BaseConnector extends HttpErrorFunctions {
 
   implicit class RequestBuilderHelpers(requestBuilder: RequestBuilder) {
 
-    def executeAccepted(implicit ec: ExecutionContext): Future[Unit] =
+    private def expectStatus(expected: Int)(implicit ec: ExecutionContext): Future[Unit] =
       requestBuilder
         .execute[HttpResponse]
         .flatMap {
           response =>
             response.status match {
-              case ACCEPTED => Future.successful(())
-              case _        => response.error
+              case `expected` => Future.successful(())
+              case _          => response.error
             }
         }
 
+    def executeAccepted(implicit ec: ExecutionContext): Future[Unit] = expectStatus(ACCEPTED)
```

Comment:
Rather than creating a function for each status code, would it make sense to make expectStatus public and call that instead?